src/code/z_lights.c

```diff
@@ -68,16 +68,13 @@ void Lights_Draw(Lights* lights, GraphicsContext* gfxCtx) {
     light = &lights->l.l[0];
 
     while (i < lights->numLights) {
-        i++;
-        gSPLight(POLY_OPA_DISP++, light, i);
+        gSPLight(POLY_OPA_DISP++, light, ++i);
         gSPLight(POLY_XLU_DISP++, light, i);
         light++;
     }
 
-    if (0) {}
-
-    i++; // ambient light is total number of lights + 1
-    gSPLight(POLY_OPA_DISP++, &lights->l.a, i);
+    // ambient light is total number of lights + 1
+    gSPLight(POLY_OPA_DISP++, &lights->l.a, ++i);
```

Comment:
Is this required for matching? We try to keep incrementing separate as much as possible.

Reply:
In isolation, no. It does work on it's own line + `if (0)` (as it was), or you can move it to the same line and get rid of the `if (0)`. Pick your poison, but I can revert this if what it was before is preferred

Reply:
Yeah, that's not really any better (and -g3 might like it even less).

Reply:
Actually, I found this match because I did a lot of comparisons to MM for this PR, and this particular function had `++i` in `gSPLight(`

Reply:
It would be better if the increment could stay out but if it can't well so be it